app/lib/features/chat_ng/widgets/events/reply_original_event.dart

```diff
@@ -0,0 +1,160 @@
+import 'package:acter/common/providers/common_providers.dart';
+import 'package:acter/common/providers/room_providers.dart';
+import 'package:acter/features/chat/widgets/messages/encrypted_message.dart';
+import 'package:acter/features/chat/widgets/messages/redacted_message.dart';
+import 'package:acter/features/chat_ng/widgets/events/file_message_event.dart';
+import 'package:acter/features/chat_ng/widgets/events/image_message_event.dart';
+import 'package:acter/features/chat_ng/widgets/events/text_message_event.dart';
+import 'package:acter/features/chat_ng/widgets/events/video_message_event.dart';
+import 'package:acter_avatar/acter_avatar.dart';
+import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart'
+    show RoomEventItem, RoomMessage;
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+import 'package:acter/common/extensions/options.dart';
+
+class ReplyOriginalEvent extends StatelessWidget {
+  final String roomId;
+  final RoomMessage msg;
+  const ReplyOriginalEvent({
+    super.key,
+    required this.roomId,
+    required this.msg,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    // reply is allowed for only EventItem not VirtualItem
+    // user should be able to get original event as RoomMessage
+    RoomEventItem replyEventItem =
+        msg.eventItem().expect('room msg should have event item');
+    final messageId = msg.uniqueId();
```

Comment:
if so, why not directly ask for both of these as the input param rather than only `msg`?